### ISSUE TITLE
Put libco, etc. in separate repositories

### ISSUE THREAD

**Issue by namandixit** [higan-emu/higan#40]
Currently, all libraries are a part of the monorepo, which makes it harder for others to use. Also, since `bsnes` and `higan` share some of these libraries, keeping these in sync is a challenge. For example, at the moment, the `libco` directory in [`higan`](https://github.com/higan-emu/higan/tree/master/libco) and [`bsnes`](https://github.com/bsnes-emu/bsnes/commits/master/libco) are out of sync.

It seems like putting these libraries in separate repositories and using either git's `subrepo` feature or mentioning the dependencies in the README might be a good idea.

**Comment by namandixit** [comment]
For the time being, I have forked the `libco` directory alone [here](https://github.com/namandixit/libco). Hopefully, maintaining this fork will not be necessary for long.

**Comment by MerryMage** [comment]
https://github.com/higan-emu/libco is now available. I only just realized you did a `filter-branch` to preserve the history, but oh well. Extracting out other libraries is planned.

**Comment by namandixit** [comment]
@MerryMage Thanks. I'll leave this issue opened in case you want to use it to track the extraction of other libraries, as well as using the new `libco` in `higan` instead of keeping the files in directory.

**Comment by Screwtapello** [comment]
> I only just realized you did a filter-branch to preserve the history...

Actually, `git subtree split` extracts all the commits that affect a particular directory tree into their own history. The opposite operation is `git subtree merge`.

**Comment by FitzRoyX** [comment]
This might be a stupid question, but I'm used to downloading the source from my browser and then compiling it as is. Do I need to now copy libco into the source folder after downloading it since it will be absent?

**Comment by Screwtapello** [comment]
No, libco is [still there](https://github.com/higan-emu/higan/tree/dd7a2baa4a3492ed9820ac3fbde34b0f094ba188/libco), it's just that now the commit history [contains tags](https://github.com/higan-emu/higan/commit/dd7a2baa4a3492ed9820ac3fbde34b0f094ba188) (all those "git-subtree" bits) that allow it to be automatically sync'd with the external repository.

**Comment by Screwtapello** [comment]
I've also split out a separate repo for nall, although I haven't yet spliced it into higan or bsnes: https://github.com/higan-emu/nall/
